compiler/src/dotty/tools/dotc/reporting/diagnostic/Message.scala

```diff
@@ -12,7 +12,7 @@ object Message {
     * not yet been ported to the new scheme. Comment out this `implicit def` to
     * see where old errors still exist
     */
-  implicit def toNoExplanation(str: String): Message =
+  implicit def toNoExplanation(str: => String): Message =
```

Comment:
It's worth noting that we cannot have implicit conversion from a by-name param when using Dotty's `Conversion` mechanism.

Reply:
Yes, I  noted that as well. That's a problem.